tests/ashley/test_auth_backend.py

```diff
@@ -0,0 +1,133 @@
+"""Test suite for ashley authentication backend."""
+
+from typing import Optional
+from urllib.parse import urlencode
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import PermissionDenied
+from django.test import RequestFactory, TestCase
+
+from ashley.auth.backend import LTIBackend
+from lti_provider.factories import LTIConsumerFactory, LTIPassportFactory
+from lti_provider.lti import LTI
+
+from tests.lti_provider.utils import CONTENT_TYPE, sign_parameters
+
+
+class LTIBackendTestCase(TestCase):
+    """Test the LTIBackend class"""
+
+    def setUp(self):
+        """Override the setUp method to instanciate and serve a request factory."""
+        super().setUp()
+        self.request_factory = RequestFactory()
+        self._consumer1 = LTIConsumerFactory(slug="test_auth_backend_consumer1")
+        self._consumer1_passport1 = LTIPassportFactory(
+            title="consumer1_passport1", consumer=self._consumer1
+        )
+
+        self._consumer2 = LTIConsumerFactory(slug="test_auth_backend_consumer2")
+        self._consumer2_passport1 = LTIPassportFactory(
+            title="consumer2_passport1", consumer=self._consumer2
+        )
+        self._auth_backend = LTIBackend()
+
+    def _authenticate(
+        self,
+        lis_person_contact_email_primary: Optional[str] = "ashley@example.com",
+        lis_person_sourcedid: Optional[str] = "ashley",
+        passport=None,
+    ):
+        if not passport:
+            passport = self._consumer1_passport1
+        lti_parameters = {
+            "lti_message_type": "basic-lti-launch-request",
+            "lti_version": "LTI-1p0",
+            "resource_link_id": "aaa",
+            "context_id": "course-v1:fooschool+authbackend+0001",
+        }
+        if lis_person_contact_email_primary:
+            lti_parameters[
+                "lis_person_contact_email_primary"
+            ] = lis_person_contact_email_primary
+        if lis_person_sourcedid:
+            lti_parameters["lis_person_sourcedid"] = lis_person_sourcedid
+
+        url = "http://testserver/lti/launch"
+        signed_parameters = sign_parameters(passport, lti_parameters, url)
+        request = self.request_factory.post(
+            url, data=urlencode(signed_parameters), content_type=CONTENT_TYPE
+        )
+        lti = LTI(request)
+        lti.verify()
+        return self._auth_backend.authenticate(request, lti_request=lti)
+
+    def test_known_user(self):
+        """Test authentication with an already existing user."""
+
+        known_user = get_user_model().objects.create_user(
+            "test_auth_backend_user1",
+            email="ashley@example.com",
+            public_username="ashley",
+            lti_consumer=self._consumer1,
+            lti_remote_user_id="ashley",
+        )
+
+        auth_user = self._authenticate()
```

Comment:
Even if those are defaults of your method, I think it's way more explicit to specify some arguments:

```suggestion
        auth_user = self._authenticate("ashley@example.com", "ashley")
```

Reply:
I've rewritten the tests to call the `_authenticate()` helper function with an exhaustive list of LTI parameters.